app/src/main/java/com/keylesspalace/tusky/adapter/AccountListAdapter.kt

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2018 Jeremiasz Nelz
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.adapter
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.RelativeLayout
+import android.widget.TextView
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.db.AccountEntity
+import com.pkmmte.view.CircularImageView
+import com.squareup.picasso.Picasso
+
+class AccountListAdapter(private val accountList: List<AccountEntity>, private val onAccountSelectedListener : OnAccountSelectedListener) : RecyclerView.Adapter<AccountListAdapter.AccountListViewHolder>() {
+
+    override fun getItemCount(): Int {
+        return accountList.size
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AccountListViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_account, parent, false) as RelativeLayout
+        return AccountListViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: AccountListViewHolder, position: Int) {
+        val account = accountList[position]
+
+        val avatar = holder.accountItem.findViewById<CircularImageView>(R.id.account_avatar)
```

Comment:
I'm sorry, but that's not how `ViewHolder` pattern works. The whole point of holders is to avoid expensive view search on every item bind. Look it up, but in general you find views in `ViewHolder` constructor and just fill in data in `onBindViewHolder`

Reply:
It has the same elements anyway, so… ¯\_(ツ)_/¯
Anyway, I'm changing this :Þ

Reply:
Well, ofc they're the same, what's the point of traversing the view hierarchy every time then?